spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,333 @@
+# IMPORTANT NOTE: The script process *only* axial slices at the level of intervertebral discs C3/C4 (value: 4) to C6/C7
+# (value: 7). In other words, the script does *not* process all axial slices of the spinal cord segmentation.
+#
+# The script requires spinal cord segmentation (used to compute the shape metrics) and disc labels.
+#
+# More details in: https://pubmed.ncbi.nlm.nih.gov/35371944/
+#
+# Copyright (c) 2024 Polytechnique Montreal <www.neuro.polymtl.ca>
+# License: see the file LICENSE
+
+import sys
+import math
+from typing import Sequence
+import textwrap
+
+import numpy as np
+import pandas as pd
+
+from spinalcordtoolbox.process_seg import compute_shape
+from spinalcordtoolbox.aggregate_slicewise import aggregate_per_slice_or_level, func_wa, func_std, merge_dict
+from spinalcordtoolbox.centerline.core import ParamCenterline
+from spinalcordtoolbox.template import get_slices_from_vertebral_levels
+from spinalcordtoolbox.image import Image, add_suffix, splitext
+from spinalcordtoolbox.utils.fs import get_absolute_path
+from spinalcordtoolbox.utils.shell import SCTArgumentParser, Metavar
+from spinalcordtoolbox.utils.sys import init_sct, set_loglevel, printv
+
+
+# Currently, the model works only for discs C3/C4 (4) to C6/C7 (7)
+SUPPORTED_DISCS = {4: 'C3/C4', 5: 'C4/C5', 6: 'C5/C6', 7: 'C6/C7'}
+# Compression cut-offs --> compression probability
+#   >0.451 high probability of compression --> yes
+#   0.345-0.451 moderate probability of compression --> possible
+#   <0.345 low probability of compression --> no
+CUT_OFF_HIGH = 0.451
+CUT_OFF_MODERATE = 0.345
+
+
+def get_parser():
+    parser = SCTArgumentParser(
+        description=textwrap.dedent("""
+            Predict compression probability in a spinal cord MRI image using spinal cord shape metrics.
+            IMPORTANT NOTE: The script process *only* axial slices at the level of intervertebral discs C3/C4 (value: 4) to C6/C7 (value: 7).
+            More details in: https://pubmed.ncbi.nlm.nih.gov/35371944/
+        """),  # noqa: E501 (line too long)
+    )
+
+    mandatoryArguments = parser.add_argument_group("MANDATORY ARGUMENTS")
+    mandatoryArguments.add_argument(
+        '-s',
+        metavar=Metavar.file,
+        required=True,
+        help="Segmentation of the spinal cord. Example: t2s_seg.nii.gz."
+    )
+    mandatoryArguments.add_argument(
+        '-discfile',
+        metavar=Metavar.file,
+        required=True,
+        help=textwrap.dedent("""
+            File with disc labels. Example: t2s_discs.nii.gz.
+            The convention for disc labels is the following: value=3 -> disc C2/C3, value=4 -> disc C3/C4, etc.
+            Such a label file can be manually created using: sct_label_utils -i IMAGE_REF -create-viewer 4:7 or
+            obtained automatically using the sct_label_vertebrae function (the file with the \'labeled_discs.nii.gz\' suffix).
+        """),  # noqa: E501 (line too long)
```

Comment:
You could specify that it can be at the posterior tip of the disc or centered on the spinal cord maybe? To be consistent with other functions like `sct_register_to_template`)

Reply:
Good point! Thanks! Done as part of https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4760/commits/ea9f82125c3aeaff0b574e619c65e4904d0e89fd!